bids/layout/layout.py

```diff
@@ -380,29 +397,30 @@ def _init_db(self, database_path=None, reset_database=False):
         # to setting the session (which creates the empty database file)
         reset_database = (
             reset_database or  # Manual Request
-            not database_path or  # In memory transient db
-            not os.path.exists(database_file)  # New file based db created
+            not database_file or  # In memory transient db
+            not database_file.exists()  # New file based db created
         )
 
         self._set_session(database_file)
 
-        instance_args = self._sanitize_instance_args()
-
         if not reset_database:
             with open(database_sidecar) as fobj:
                 saved_args = json.load(fobj)
```

Comment:
```python
saved_args = json.loads(database_sidecar.read_text())
```

Reply:
I swore I had accepted this suggestion a long time ago. I wonder if I purposely reverted this. I guess we'll see if tests pass.

Reply:
Oh, I do this in two places, that's why.